tls/s2n_record_write.c

```diff
@@ -85,6 +85,66 @@ int s2n_record_max_write_payload_size(struct s2n_connection *conn)
     return max_fragment_size - overhead(conn);
 }
 
+int s2n_record_write_encryption(struct s2n_connection *conn, const struct s2n_cipher_suite *cipher_suite, struct s2n_session_key *session_key,
+                                uint16_t block_size, struct s2n_blob iv, uint8_t *implicit_iv, struct s2n_blob en)
+{
+    switch (cipher_suite->record_alg->cipher->type) {
+        case S2N_STREAM:
+            GUARD(cipher_suite->record_alg->cipher->io.stream.encrypt(session_key, &en, &en));
+            break;
+        case S2N_CBC:
+            GUARD(cipher_suite->record_alg->cipher->io.cbc.encrypt(session_key, &iv, &en, &en));
+
+            /* Copy the last encrypted block to be the next IV */
+            if (conn->actual_protocol_version < S2N_TLS11) {
+                gte_check(en.size, block_size);
+                memcpy_check(implicit_iv, en.data + en.size - block_size, block_size);
+            }
+            break;
+        case S2N_COMPOSITE:
+            /* This will: compute mac, append padding, append padding length, and encrypt */
+            GUARD(cipher_suite->record_alg->cipher->io.comp.encrypt(session_key, &iv, &en, &en));
+
+            /* Copy the last encrypted block to be the next IV */
+            gte_check(en.size, block_size);
+            memcpy_check(implicit_iv, en.data + en.size - block_size, block_size);
+            break;
+        default:
+            S2N_ERROR(S2N_ERR_CIPHER_TYPE);
+            break;
+    }
+
+    return 0;
+}
+
+uint16_t s2n_record_skip_write_encrypted_length(struct s2n_connection *conn, const struct s2n_cipher_suite *cipher_suite, struct s2n_blob iv,
```

Comment:
If I'm understanding it correctly, the goal of this function is to return the encrypted length. If that's the case, could we remove the "skip_write" from the name?